blockchain/v2/scheduler.go

```diff
@@ -547,9 +569,16 @@ func (sc *scheduler) handleNoBlockResponse(event bcNoBlockResponse) (Event, erro
 }
 
 func (sc *scheduler) handleBlockProcessed(event pcBlockProcessed) (Event, error) {
+	// No such peer or peer was removed.
+	peer, ok := sc.peers[event.peerID]
+	if !ok || peer.state == peerStateRemoved {
+		return noOp, nil
+	}
```

Comment:
Are we sure this is correct? If a block was successfully applied, shouldn't we just mark it as processed regardless of whether the peer has been removed in the meanwhile?

Reply:
I think you're right! At this point we've upgraded the state, so the only way back is to revert it which we can't do currently. Changed